src/Microsoft.AspNetCore.Razor.Language/TagHelperBinder.cs

```diff
@@ -74,11 +74,20 @@ public TagHelperBinder(string tagHelperPrefix, IEnumerable<TagHelperDescriptor>
             }
 
             var tagNameWithoutPrefix = _tagHelperPrefix != null ? tagName.Substring(_tagHelperPrefix.Length) : tagName;
+
+            var parentTagNameWithoutPrefix = parentTagName;
+            if (!string.IsNullOrEmpty(_tagHelperPrefix) &&
+                !string.IsNullOrEmpty(parentTagName) &&
+                parentTagName.Length > _tagHelperPrefix.Length)
+            {
+                parentTagNameWithoutPrefix = parentTagName.Substring(_tagHelperPrefix.Length);
```

Comment:
parent tag might not be a TagHelper. I'd imagine this would break in the following scenario:
```HTML
@tagHelperPrefix th

<parent>
    <th:requires-parent-as-parent>
</parent>
```